### PR TITLE
Write an install signature during manual agent installations

### PR DESCRIPTION
**Pull request by yshapiro-57**
### What does this PR do?

This PR makes `yum install datadog-agent` and `apt-get install datadog-agent` write an `install.json` file to the same directory as the Agent config file, `datadog.yaml`. The `install.json` file contains a random UUID, an installation timestamp, and the mechanism that was used to install the agent (the hardcoded string `linux_manual_with_apt` or `linux_manual_with_yum`). The agent will send this information to the Datadog backend, so that we can gather telemetric data on how long users take to onboard after the initial installation of the agent.

### Motivation

Jira: [Write an install signature during yum and apt-get installs](https://datadoghq.atlassian.net/browse/AIT-9866) . This is part of our broader work on improving the APM onboarding experience.

### Additional Notes

The code is copied and pasted from the [Linux agent install script](https://github.com/DataDog/agent-linux-install-script/blob/b71a6d3a46e1ecbf65e8b19c36db439d4609fbcd/install_script.sh.template#L1417-L1453)

### Possible Drawbacks / Trade-offs

I am not copying these changes to the `agent-dmg`, `iot-agent-deb` or `iot-agent-rpm` directories, because we are initially focusing on improving the onboarding experience on Linux.

## Quick test
I copied the code to a standalone shell script, ran it, and verified that an `install.json` file with the right contents gets written to my config dir:

```
> mkdir tmp
> CONFIG_DIR=$HOME/tmp
> source test.sh 
> ls tmp
install.json
> cat tmp/install.json 
{"install_id":"e8b7ba30-2b3e-484f-a935-c859dc7d02e3","install_type":"linux_manual_with_apt","install_time":1708633404}
>
```

## End to end test
To test the agent-rpm changes, I pulled up [the agent_rpm-x4-a7 CI job for this PR](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/442323573) and clicked "Download" under "Job artifacts". I unzipped artifacts.zip and copied the `omnibus/pkg/datadog-agent-7.52.0~rc.2.git.6.92bc9b7.pipeline.28849709-1.x86_64.rpm` from that file onto a brand new Amazon EC2 instance that I created in our AWS staging account just for this test. Then I SSHed to that instance and ran this command:

```
sudo yum localinstall datadog-agent-7.52.0~rc.2.git.6.92bc9b7.pipeline.28849709-1.x86_64.rpm
```
I verified that this causes an `install.json` file with the right contents to be written to `/etc/datadog-agent`
```
[ec2-user@ip-172-31-32-207 ~]$ ls /etc/datadog-agent/
checks.d      conf.d                install.json  runtime-security.d           selinux
compliance.d  datadog.yaml.example  install_info  security-agent.yaml.example  system-probe.yaml.example
[ec2-user@ip-172-31-32-207 ~]$ cat /etc/datadog-agent/install.json
{"install_id":"4db2464e-4d22-4358-ab17-eee6bf5d111b","install_type":"linux_manual_with_rpm","install_time":1708875361}
[ec2-user@ip-172-31-32-207 ~]$ sudo yum localinstall datadog-agent-7.52.0~rc.2.git.6.92bc9b7.pipeline.28849709-1.x86_64.rpm
```
I then uninstalled the agent manually with `sudo yum remove datadog-agent` and verified that `install.json` is deleted.

Repeating the test for agent-deb, I pulled up the [agent_deb-x4-a7 CI job for this PR](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/442323558) and clicked "Download" under "Job artifacts". I unzipped artifacts.zip and copied `omnibus/pkg/datadog-agent_7.52.0\~rc.2.git.6.92bc9b7.pipeline.28849709-1_amd64.deb` onto a brand new Google Cloud Linux VM instance that I created using [this link](https://console.cloud.google.com/compute/instancesAdd?project=datadog-sandbox) - I could have instead created a Debian instance in AWS. I logged in to that instance and ran
```
sudo apt-get install ./datadog-agent_7.52.0~rc.2.git.6.92bc9b7.pipeline.28849709-1_amd64.deb
```
As with rpm, I verified that this causes an `install.json` file with the right contents to be written to `/etc/datadog-agent`:
```
yakov_shapiro_datadoghq_com@instance-20240222-183827:~$ ls /etc/datadog-agent/
checks.d  compliance.d  conf.d  datadog.yaml.example  install.json  install_info  runtime-security.d  security-agent.yaml.example  selinux  system-probe.yaml.example
yakov_shapiro_datadoghq_com@instance-20240222-183827:~$ cat /etc/datadog-agent/install.json 
{"install_id":"ae2dbc27-f2aa-45f6-9f33-cbe1ff954d73","install_type":"linux_manual_with_apt","install_time":1708876392}
yakov_shapiro_datadoghq_com@instance-20240222-183827:~$
```
I then uninstalled the agent with `sudo apt-get remove datadog-agent` and verified that the `install.json` file is deleted.

### CODE REVIEW

File: omnibus/package-scripts/updater-deb/postinst [hunk 1]
```diff
@@ -77,6 +77,45 @@ install_method:
...
+
+function generate_install_signature() {
+  local install_id="$1"
+  local install_type="$2"
+  local install_time="$3"
+  printf "{\"install_id\":\"$install_id\",\"install_type\":\"$install_type\",\"install_time\":$install_time}"
+}
+
+install_id=$(generate_install_id)
+install_type="linux_manual_with_apt"
```
Comment (spencergilbert) [inline omnibus/package-scripts/updater-deb/postinst:100]: **💭 thought:** ‏This likely should be different to track the usage of the newer updating product.

File: omnibus/package-scripts/agent-rpm/posttrans [hunk 2]
```diff
@@ -108,9 +108,48 @@ install_method:
   installer_version: $installer_version
 "
     echo "$install_info_content" > $CONFIG_DIR/install_info
-    chown -R dd-agent:dd-agent ${CONFIG_DIR}
```
Comment (spencergilbert) [inline omnibus/package-scripts/agent-rpm/posttrans:111]: **❓ question:** ‏Why was this `chown` removed?

**Comment by yshapiro-57** [comment]
@spencergilbert yes, it is possible to rename the install_type in a future agent release. The install_type string is opaque to the code that processes it. It eventually makes it into Metabase, our BI database built on top of Snowflake, and it is there that we write reports on how long customers take to onboard by install type. So the only cost to changing the install type is that we would need to make the same change to those reports, which is quick and easy.

Also which owner did you have in mind? Is there another team we should loop in on this PR?

**Comment by spencergilbert** [comment]
> Also which owner did you have in mind? Is there another team we should loop in on this PR?

The updater product is new and should be owned by fleet automation. If it's trivial to update down the line I'm not sure it's worth blocking, but if we have to time to double check they're in the loop that would be nice.

**Review by spencergilbert — APPROVED** [review]

**Review by raphaelgavache — APPROVED** [review]
lgtm for updater install script

**Pull request merged**